src/operator/random/shuffle_op.cc

```diff
@@ -55,18 +56,24 @@ void Shuffle1D(DType* const out, const index_t size, Rand* const prnd) {
 
 template<typename DType, typename Rand>
 void ShuffleND(DType* const out, const index_t size, const index_t first_axis_len,
-                Rand* const prnd) {
+                Rand* const prnd, const OpContext& ctx) {
   // Fisher-Yates shuffling
+  using namespace mxnet_op;
   const index_t stride = size / first_axis_len;
   auto rand_n = [prnd](index_t n) {
     std::uniform_int_distribution<index_t> dist(0, n - 1);
     return dist(*prnd);
   };
   CHECK_GT(first_axis_len, 0U);
+  const size_t stride_bytes = sizeof(DType) * stride;
+  Tensor<cpu, 1, char> buf =
+    ctx.requested[1].get_space_typed<cpu, 1, char>(Shape1(stride_bytes), ctx.get_stream<cpu>());
```

Comment:
I think you need to add Fresources to operator registration?

Reply:
It is [there](https://github.com/apache/incubator-mxnet/blob/5a480f0237c28231bcf486f39f917b65513ee6a3/src/operator/random/shuffle_op.cc#L124) because the GPU version already needed a temp space.